File: src/com/fenghuo/notes/bean/Bean.java
package com.fenghuo.notes.bean;

import com.google.gson.Gson;

/**
 * Created by gang on 16-4-28.
 */
public class Bean {

    private static Gson gson;

    static {
        gson = new Gson();
    }

    public final String toJsonString() {
        return gson.toJson(this);
    }

    public static <T extends Bean> T readFromJson(String json, Class<T> clazz) {
        return gson.fromJson(json, clazz);
    }

}

File: src/com/fenghuo/notes/bean/Bean.kt
package com.fenghuo.notes.bean

import com.google.gson.Gson

/**
 * Created by gang on 16-4-28.
 */
open class Bean {

    fun toJsonString(): String {
        return gson!!.toJson(this)
    }

    companion object {

        private var gson: Gson? = null

        init {
            gson = Gson()
        }

        fun <T : Bean> readFromJson(json: String, clazz: Class<T>): T {
            return gson!!.fromJson(json, clazz)
        }
    }

}
